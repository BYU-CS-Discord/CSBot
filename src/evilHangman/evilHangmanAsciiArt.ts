import fs from 'node:fs/promises';

const ART_PATH = './res/hangmen.txt';
const FRAME_SEPARATOR = 'Ã¦';

async function getFrames(): Promise<ReadonlyArray<string>> {
	const file = await fs.readFile(ART_PATH);
	const allFrames = file.toString();
	return allFrames.split(FRAME_SEPARATOR);
}

export async function getHangmanArt(guessesRemaining: number, outOf: number): Promise<string> {
	const frames = await getFrames();
	const scale = (frames.length - 1) / outOf;
	const frameIndex = Math.floor(frames.length - guessesRemaining * scale - 1);
	return frames[frameIndex] ?? '';
}
