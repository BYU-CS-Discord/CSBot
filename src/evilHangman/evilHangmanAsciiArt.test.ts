import { describe, expect, test } from 'vitest';

import { getHangmanArt } from './evilHangmanAsciiArt';

describe('evilHangmanAsciiArt.ts', () => {
	const tests = [
		[0, 1],
		[0, 7],
		[3, 7],
		[4, 7],
		[7, 7],
		[50, 1000],
	] as const;
	test.each(tests)('frame %i of %i has not changed', async (frame, ofFrame) => {
		await expect(getHangmanArt(frame, ofFrame)).resolves.toMatchSnapshot();
		await expect(getHangmanArt(frame, ofFrame)).resolves.not.toBe('');
	});

	test('out of bounds resolves to empty string', async () => {
		await expect(getHangmanArt(11, 10)).resolves.toBe('');
	});
});
