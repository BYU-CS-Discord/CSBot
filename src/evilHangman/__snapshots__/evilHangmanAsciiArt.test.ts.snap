// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evilHangmanAsciiArt.ts > frame 0 of 1 has not changed 1`] = `
"
 /---\\
 o   |
\\|/  |
 |   |
/ \\  |
    -|-
"
`;

exports[`evilHangmanAsciiArt.ts > frame 0 of 7 has not changed 1`] = `
"
 /---\\
 o   |
\\|/  |
 |   |
/ \\  |
    -|-
"
`;

exports[`evilHangmanAsciiArt.ts > frame 3 of 7 has not changed 1`] = `
"
 /---\\
 o   |
\\|/  |
     |
     |
    -|-
"
`;

exports[`evilHangmanAsciiArt.ts > frame 4 of 7 has not changed 1`] = `
"
 /---\\
 o   |
\\|   |
     |
     |
    -|-
"
`;

exports[`evilHangmanAsciiArt.ts > frame 7 of 7 has not changed 1`] = `
" /---\\
     |
     |
     |
     |
    -|-
"
`;

exports[`evilHangmanAsciiArt.ts > frame 50 of 1000 has not changed 1`] = `
"
 /---\\
 o   |
\\|/  |
 |   |
/    |
    -|-
"
`;
